src/cpp/dataset.cpp

```diff
@@ -378,6 +379,22 @@ inline void DataSet::_enforce_tensor_exists(const std::string& tensorname) const
     }
 }
 
+// Check that the provided tensor type matches the internal tensor type
+inline void DataSet::_enforce_tensor_type(const std::string& tensorname,
+                                          const SRTensorType& type) const
+{
+    _enforce_tensor_exists(tensorname);
+
+    SRTensorType known_type = _tensorpack.get_tensor(tensorname)->type();
```

Comment:
I'm a little rusty on how `get_tensor` is implemented, so wanted to double check. IIRC it just returns a pointer to the existing tensor, so the extra additional call to `get_tensor` has a negligible impact on performance, correct?

Reply:
Yes, that is correct.  The underlying data structure is ``std::unordered_map<std::string, TensorBase*>`` and we are just doing an ``at`` call on that unordered_map (getting a pointer).